packages/remix-testing/create-remix-stub.tsx

```diff
@@ -42,7 +52,55 @@ type RemixStubOptions = {
   remixConfigFuture?: Partial<FutureConfig>;
 };
 
-export function createRemixStub(routes: (RouteObject | DataRouteObject)[]) {
+function patchRoutesWithContext(
+  routes: (StubRouteObject | StubDataRouteObject)[],
+  context: AppLoadContext
+): (RouteObject | DataRouteObject)[] {
+  return routes.map((route) => {
+    if (route.children) {
+      return {
+        ...route,
+        children: patchRoutesWithContext(route.children, context),
+      };
+    }
```

Comment:
Should this come after the loader/action patch so we don't miss patching the parent's loader/action?

Reply:
yep, had it that way originally, reset and then copilot decided it was to be first 😅